httpx/_utils.py

```diff
@@ -123,20 +123,27 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
 
 
 class NetRCInfo:
-    def __init__(self, files: typing.Optional[typing.List[str]] = None) -> None:
+    def __init__(self, files: typing.Optional[typing.List[str]] = None, raise_errors = False) -> None:
         if files is None:
             files = [os.getenv("NETRC", ""), "~/.netrc", "~/_netrc"]
         self.netrc_files = files
+        self.raise_errors = raise_errors
 
     @property
     def netrc_info(self) -> typing.Optional[netrc.netrc]:
         if not hasattr(self, "_netrc_info"):
             self._netrc_info = None
             for file_path in self.netrc_files:
                 expanded_path = Path(file_path).expanduser()
-                if expanded_path.is_file():
-                    self._netrc_info = netrc.netrc(str(expanded_path))
-                    break
+                try:
+                    if expanded_path.is_file():
+                        self._netrc_info = netrc.netrc(str(expanded_path))
+                        break
+                except (netrc.NetrcParseError, IOError, PermissionError):
+                    # If there was a parsing error or a permissions issue reading the file,
+                    # we'll just skip netrc auth unless explicitly asked to raise errors.
+                    if self.raise_errors:
+                        raise
```

Comment:
Given that we don't actually allow passing `raise_errors=True` anywhere, I'd suggest we drop this parameter and just do:

```suggestion
                    # Issue while reading the netrc file, ignore...
                    pass
```